### PR TITLE
Use webpack's handlebars loader instead of compiling all templates

### PR DESCRIPTION
**Pull request by pattiereaves**
Instead of compiling and saving all templates, this uses the handlebars and json loaders to render the files on the fly.

Benefits:
- More explicit what the template is/does
- Easier to transfer Handlebars templates over to PHP instead of rendered files.

In this PR:
- Adds chalk to colorize terminal output.
- Moves the 'templates' field in the Huron config to 'output' (since it's where the files will be output). Change all references from `huron.templates` to `huron.output`.
- Refactors actions.js so that when it encounters hbs/json files, it will copy them over instead of compiling them and build a object in the templates array referencing those files.
- Create a reference to a templates loader in the config (and move it over using generate config).
- Add a JSON loader.
- In require-templates, if it comes to an object instead of a path, require each property in that object individually.
- Refactor cycleEl so that if the template match is an object (referencing the data/template paths) then use the callback function from the config to render it and replace it in the rendered prototype.
  - Also look for a data attribute on the tag to replace for huronState which links it to JSON object.
- Adds documentation for what needs to be done to get HMR working again.

🙅 Not totally on board with merging this yet, more putting this up here for discussion. To discuss: finishing the HMR issues before merging?

### CODE REVIEW

**Comment by pattiereaves** [comment]
Corresponding Dev-Retreat-Huron PR: https://github.com/alleyinteractive/dev-retreat-huron/pull/3

File: src/cli/actions.js [hunk 1]
```diff
@@ -96,37 +99,31 @@ export function updateFile(filepath, sections, templates, huron) {
...
           } else {
             reject(`Rejected file: ${file.name}`);
           }
         }
         break;
 
       // Handlebars template, external
-      case '.hbs':
-      case '.handlebars':
+      case huron.templates.extension:
```
Comment (pattiereaves) [inline src/cli/actions.js]: :) :) :)

File: src/cli/require-templates.js [hunk 2]
```diff
@@ -8,99 +8,52 @@ export default function requireTemplates(huron, templates, sections) {
...
+      const newModules = newAssets.keys()
+        .map((key) => {
+          return [key, newAssets(key)];
+        })
+        .filter((newModule) => {
+          return modules[newModule[0]] !== newModule[1];
+        });
+
+      newModules.forEach((module) => {
+        modules[module[0]] = module[1];
```
Comment (pattiereaves) [inline src/cli/require-templates.js]: I don't really follow how module0 = module1 but this is pretty cool.

Comment (ostowe) [inline src/cli/require-templates.js]: `module[0]` = Relative require path (so something like `./partials/my-template.hbs`). `modules` is an object in which the keys are require paths like that, and the values are the modules themselves.
`module[1]` = the contents of the module itself.

So this line here will update both the module path (or filename, or whatever) and the module contents to make sure the whole list of modules is up-to-date.

**Comment by ostowe** [comment]
merging this so we can use if for Saturday 👍

**Pull request merged**